src/common/utils/filesystem.cc

```diff
@@ -34,13 +35,15 @@ absl::StatusOr<std::filesystem::path> CreateTemporaryDirectory() {
         "Unable to access temporary file directory: %s", error_code.message()));
   }
 
-  absl::string_view temp_file = std::tmpnam(nullptr);
-  std::filesystem::path new_temp_path = temp_path / temp_file;
-  if (!std::filesystem::create_directory(new_temp_path, error_code)) {
+  const int BUFFER_SIZE = 1024;
```

Comment:
This feels old-school but I looked at `string_view` and `std::string` and in the end since `mkdtemp` needs a non-const `char *`, whatever we do we need to build that, and this seemed not worse than other approaches, but I welcome alternatives.

It also seems silly there isn't a cleaner C++ filesystem lib way to create a temporary directory with a known unique name, but this seems to be the thing, or at least one version of it, again feedback welcome.

The complaint seems to come from `g++` so we could also maybe turn it off, but one presumes it's well intentioned.

Reply:
The warning on `tmpnam` comes from the fact that the operating system tries really hard to guess a unique name, but it cannot guarantee that it truly is unique nor that someone else just happened to call `tmpnam` or similar and create that name between when you call `tmpnam` and when you use it. In general, that opens the door to TOCTOU attacks.

Now, TOCTOU attacks are bad, but I think this might not fix the underlying issue. We are only using the temporary directory so that we can communicate with Souffle by handing it a directory with input facts and reading the output facts from that directory. Because we are creating the directory so that we can, at a later point, read the output of some other program that was placed in files in that directory, the TOCTOU window is probably wide open regardless of how atomically the directory itself is created. If some other program accidentally or maliciously places `.fact` files over the Souffle output, we would trustingly read them as if they were the Souffle output files.

This solution does make the creation of the directory atomic, but it relies upon unix-specific functions (`mkdtemp`), and so wouldn't work on other systems (say, a Windows box) while this less-atomic system is more portable.

In the future, we'd like to talk to Souffle entirely through an in-memory interface, which would sidestep the TOCTOU concerns (and probably be a lot faster too). We'd likely eliminate the temporary directory entirely, or use it only for debugging. So I'm not sure if this change is worth the portability and readability cost.

Reply:
Rewrote to use `absl::StrFormat`.